src/any_agent/frameworks/langchain.py

```diff
@@ -132,12 +133,16 @@ async def run_async(self, prompt: str, **kwargs: Any) -> "AgentTrace":
         if not self._agent:
             error_message = "Agent not loaded. Call load_agent() first."
             raise ValueError(error_message)
-        self._setup_tracing()
         inputs = {"messages": [("user", prompt)]}
-        result = await self._agent.ainvoke(inputs, **kwargs)
+        run_id = str(uuid4())
+        tracer = self._tracer_provider.get_tracer("any_agent")
+        with tracer.start_as_current_span("agent_run") as span:
+            span.set_attribute("any_agent.run_id", run_id)
+            result = await self._agent.ainvoke(inputs, **kwargs)
```

Comment:
Here's how this works:

1. Make run UUID
2. Make all spans created by ainvoke be associated with the agent_run span, with which we attached the run_id attribute
3. The exporter will see this span and then have the info we need to associate the run_id (which the Agent knows about) with the trace_id (that the Agent doesn't know about)
4. Pop that trace out of the exporter so that we free up the memory since we don't need it anymore once the agent retrieves it.

Reply:
Do we need to create the parent span ourselves?

Afaik, openinference will (correctly?) create a parent span (ideally, type "Agent") already with an unique ID. 

In our exporter, we could just:

- If the trace doesn't have parent id -> Store it in `run_trace_mapping`.
- If the trace has a parent id -> Get it from `run_trace_mapping`.

I am missing something?

Reply:
If we don't create the parent span ourselves, how do we set the run_id attribute that we need so that the agent knows which trace_id is associated with the run?